eth/query/block.go

```diff
@@ -79,14 +79,28 @@ func GetBlockByNumber(
 	blockInfo.Number = eth.EncInt(height)
 	bloomFilter := evmAuxStore.GetBloomFilter(uint64(height))
 	blockInfo.LogsBloom = eth.EncBytes(bloomFilter)
+	var blockResults *ctypes.ResultBlockResults
+	if full {
+		blockResults, err = blockStore.GetBlockResults(&height)
+		if err != nil {
+			return resp, errors.Wrapf(err, "results for block %v", height)
+		}
```

Comment:
An error here means the `blockResults` couldn't be loaded, if you return immediately then `blockResults == nil` always be `false` and the code will never fall back to using `blockStore.GetTxResult(tx.Hash())`.